src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -873,6 +844,113 @@ internal int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancella
             return Succeeded;
         }
 
+        private bool CompileAndEmitRefAssembly(Compilation compilation, EmitOptions refEmitOptions, string finalRefPeFilePath,
+            TouchedFileLogger touchedFilesLogger, DiagnosticBag diagnosticBag, CancellationToken cancellationToken)
+        {
+            var refModuleBeingBuilt = compilation.CheckOptionsAndCreateModuleBuilder(
+                diagnosticBag,
+                Arguments.ManifestResources,
+                refEmitOptions,
+                debugEntryPoint: null,
+                sourceLinkStream: null,
+                embeddedTexts: null,
+                testData: null,
+                cancellationToken: cancellationToken);
+
+            bool success = true;
+            if (refModuleBeingBuilt != null)
+            {
+                try
+                {
+                    success = compilation.CompileMethods(
+                        refModuleBeingBuilt,
+                        emittingPdb: false,
+                        diagnostics: diagnosticBag,
+                        filterOpt: null,
+                        cancellationToken: cancellationToken);
+                }
+                finally
+                {
+                    refModuleBeingBuilt.CompilationFinished();
+                }
+
+                if (success)
+                {
+                    var refPeStreamProvider = new CompilerEmitStreamProvider(this, finalRefPeFilePath);
+
+                    try
+                    {
+                        success = compilation.SerializeToPeStream(
+                            refModuleBeingBuilt,
+                            refPeStreamProvider,
+                            pdbStreamProvider: null,
+                            testSymWriterFactory: null,
+                            diagnostics: diagnosticBag,
+                            metadataOnly: true,
+                            cancellationToken: cancellationToken);
+                    }
+                    finally
+                    {
+                        refPeStreamProvider.Close(diagnosticBag);
+                    }
+
+                    if (success && touchedFilesLogger != null)
+                    {
+                        touchedFilesLogger.AddWritten(finalRefPeFilePath);
+                    }
+                }
+            }
+            return success;
```

Comment:
>success [](start = 19, length = 7)

It looks like we are returning success if refModuleBeingBuilt is null. Is this intentional? Consider adding a comment.